paasta_tools/kubernetes/bin/kubernetes_remove_evicted_pods.py

```diff
@@ -0,0 +1,155 @@
+# limitations under the License.
+"""
+Usage: ./kubernetes_remove_evicted_pods.py [options]
+
+Removes Evicted pods and notifies service owners
+"""
+import argparse
+import logging
+from collections import namedtuple
+from typing import List
+from typing import Mapping
+from typing import Sequence
+
+from kubernetes.client import V1DeleteOptions
+from pysensu_yelp import Status
+
+from paasta_tools.kubernetes_tools import get_all_pods
+from paasta_tools.kubernetes_tools import KubeClient
+from paasta_tools.kubernetes_tools import V1Pod
+from paasta_tools.monitoring_tools import send_event
+from paasta_tools.utils import DEFAULT_SOA_DIR
+
+
+log = logging.getLogger(__name__)
+EvictedPod = namedtuple("EvictedPod", ["podname", "namespace", "eviction_msg"])
+
+
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser(
+        description="Removes evicted pods and notifies service owners"
+    )
+    parser.add_argument(
+        "-d",
+        "--soa-dir",
+        dest="soa_dir",
+        default=DEFAULT_SOA_DIR,
+        help="define a different soa config directory",
+    )
+    parser.add_argument(
+        "-v", "--verbose", action="store_true", dest="verbose", default=False
+    )
+    parser.add_argument(
+        "-n", "--dry-run", action="store_true", dest="dry_run", default=False
+    )
+    args = parser.parse_args()
+    return args
+
+
+def get_evicted_pods(pods: Sequence[V1Pod]) -> Sequence[V1Pod]:
+    return [
+        pod
+        for pod in pods
+        if pod.status.phase == "Failed" and pod.status.reason == "Evicted"
+    ]
+
+
+def get_pod_service(pod: V1Pod) -> str:
+    return pod.metadata.labels.get("paasta.yelp.com/service")
+
+
+def notify_service_owners(
+    services: Mapping[str, List[EvictedPod]], soa_dir: str, dry_run: bool
+) -> None:
+    check_overrides = {
+        "page": False,
+        "alert_after": "0m",
+        "realert_every": 1,
+        "tip": "Pods can be Evicted if they go over the allowed quota for a given resource. Check the Eviction message to figure out which resource quota was breached",
+    }
+    for service in services.keys():
+        check_name = f"pod-eviction.{service}"
+        check_output = "The following pods have been evicted and will be removed from the cluster:\n"
+        for pod in services[service]:
+            check_output += f"- {pod.podname}: {pod.eviction_msg}\n"
+        if dry_run:
+            log.info(f"Would have notified owners for service {service}")
+        else:
+            log.info(f"Notifying owners for service {service}")
+            send_event(
+                service,
+                check_name,
+                check_overrides,
+                Status.CRITICAL,
+                check_output,
+                soa_dir,
+            )
+
+
+def remove_pods(
+    client: KubeClient, services: Mapping[str, List[EvictedPod]], dry_run: bool
+) -> None:
+    delete_options = V1DeleteOptions()
```

Comment:
nit
Deletions are synchronous by default, which means that the script will be blocked untime the pod will be successfully deleted.
May be use `Background` to speed up the script?

Reply:
Sure! I will add the Background propagation_policy and update the review in a moment

Reply:
It seems that you can also set `gracefull_period_seconds` to zero
See https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/V1DeleteOptions.md

Reply:
yeah, that param is already set on `delete_namespaced_pod`.
Do you think that it needs to be set on V1DeleteOptions as well?

Reply:
anything that works fast and correctly is fine for me :)